### PR TITLE
[squeezebox] Add support for LMS authentication

### LINKED ISSUE #2334: [SqueezeBox] not able to login with user/password
The Squeezebox-binding ist not able to use user/password.

### PR DESCRIPTION
**Pull request by mhilbush**
Signed-off-by: Mark Hilbush <mark@hilbush.com>

Fixes #2334 : Add support for authenticating to LMS (squeeze server) using userId and password
- add userId and password config parameters to squeeze server thing
- log in to LMS if userId and password are populated; otherwise, function as before
- adjust squeeze server discovery to use default CLI port (9090) on `401 Unauthorized` response

Fix code analysis issues

Update README for LMS authentication

Update README known issues list for squeezelite and piCorePlayer

### CODE REVIEW

File: addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java [hunk 1]
```diff
@@ -56,14 +58,20 @@ public static String post(String url, String postData) throws Exception {
...
-                .timeout(TIMEOUT, TimeUnit.MILLISECONDS)
+                .timeout(timeout, TimeUnit.MILLISECONDS)
                 .send();
 
         int statusCode = response.getStatus();
 
+        if (statusCode == HttpStatus.UNAUTHORIZED_401) {
+            String statusLine = response.getStatus() + " " + response.getReason();
+            logger.error("Received '{}' from squeeze server", statusLine);
+            throw new NotAuthorizedException("Unauthorized: " + statusLine);
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java]: Maybe it's better not to depend on this exception but create your own this one comes from some specific Java web services parts and looks a little misplaced in the dependencies.

File: addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java [hunk 2]
```diff
@@ -33,7 +35,7 @@
 public class HttpUtils {
     private static Logger logger = LoggerFactory.getLogger(HttpUtils.class);
 
-    private static int TIMEOUT = 5000;
+    private static int timeout = 5000;
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java]: A little background, maybe you are aware, the existing thing was almost a constant in java. The only thing to complete it was adding a `final` modifier. You could consider adding the missing `final` to the class level (almost) constants and keep them uppercase.

Comment (martinvw) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java]: Because it makes sense to see a timeout as a constant.

File: addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java (hunk 1 above)
Comment (mhilbush) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java]: I was wondering about that, although this one fit the need nicely.  Not much more work to create my own, so I'll do that.

File: addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java (hunk 2 above)
Comment (mhilbush) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java]: Yes, I could've done that on this one.  Easy change.

File: addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/discovery/SqueezeBoxServerDiscoveryParticipant.java [hunk 3]
```diff
@@ -37,7 +39,7 @@
     /**
      * Name of a Squeeze Server
      */
-    private static String MODEL_NAME = "Logitech Media Server";
+    private static String modelName = "Logitech Media Server";
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/discovery/SqueezeBoxServerDiscoveryParticipant.java]: I suppose the same constant story should apply here

File: addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java (hunk 1 above)
Comment (martinvw) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java]: It's not black-and-white, I was especially triggered by the dependency to that javax package.

Comment (mhilbush) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/utils/HttpUtils.java]: > It's not black-and-white

Yea, I know.  But, before I used it, I searched the repos for any use of that exception, and there was none.  So I figured it might be questionable.

**Comment by mhilbush** [comment]
@martinvw I've pushed some changes that address your comments.

File: addons/binding/org.openhab.binding.squeezebox/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -8,6 +8,7 @@ Bundle-RequiredExecutionEnvironment: JavaSE-1.8
 Bundle-ClassPath: .
 Import-Package: com.google.common.collect,
  com.google.gson,
+ javax.ws.rs,
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.squeezebox/META-INF/MANIFEST.MF]: Sorry this import should the also be removed

File: addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/discovery/SqueezeBoxServerDiscoveryParticipant.java [hunk 5]
```diff
@@ -23,21 +23,23 @@
...
+ * @author Mark Hilbush - Add support for LMS authentication
  *
  */
 public class SqueezeBoxServerDiscoveryParticipant implements UpnpDiscoveryParticipant {
 
     /**
      * Name of a Squeeze Server
      */
-    private static String MODEL_NAME = "Logitech Media Server";
+    private static final String MODELNAME = "Logitech Media Server";
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/discovery/SqueezeBoxServerDiscoveryParticipant.java]: While you have to do another commit please restore the underscore

**Comment by mhilbush** [comment]
Removed import and restored underscore to MODELNAME.

**Comment by martinvw** [comment]
Thanks for your quick responses and fixing the code analysis issues!
